src/runtime/runtime.ts

```diff
@@ -254,20 +254,13 @@ export abstract class Runtime {
   }
 
   private makeSubAccount(name: string): string {
-    return `${name}.${this.masterAccount}`;
+    return this.root.makeSubAccount(name);
   }
 
-  async createAccount(name: string, keyPair?: nearAPI.utils.key_pair.KeyPair): Promise<Account> {
-    const accountId = this.makeSubAccount(name);
-    const pubKey = await this.addKey(accountId, keyPair);
-    await this.root.najAccount.createAccount(
-      accountId,
-      pubKey,
-      new BN(this.config.initialBalance!)
-    )
-    const account = this.getAccount(name);
-    this.accountsCreated.set(accountId, name);
-    return account;
+  async createAccount(name: string, {keyPair,initialBalance = this.config.initialBalance!}: {keyPair?: KeyPair, initialBalance?: string} = {}): Promise<Account> {
+    const newAccount = await this.root.createAccount(name, {keyPair, initialBalance});
+    this.accountsCreated.set(newAccount.accountId, name);
```

Comment:
Is it worth adding some sort of hook from `Account` back to `Runtime.afterAccountCreated` (or something, very much not into this name yet) **before** we merge, so people don't have to use `Runtime.createAccount` in order for the accounts to be tracked? Since we're not actually doing any account cleanup yet, maybe it doesn't matter, but we should definitely do this soon.

Reply:
Yeah perhaps it's a good point to pass Runtime to the account to track the accounts.